org.civicrm.reqangular/src/common/directives/h-scroll-shadows.directive.js

```diff
@@ -0,0 +1,75 @@
+/* eslint-env amd */
+/* globals MutationObserver */
+
+define([
+  'common/angular',
+  'common/modules/directives'
+], function (angular, directives) {
+  'use strict';
+
+  directives.directive('hScrollShadows', [function () {
+    return {
+      restrict: 'A',
+      controller: hScrollShadowsController
+    };
+  }]);
+
+  hScrollShadowsController.$inject = ['$element', '$scope', '$window'];
+
+  function hScrollShadowsController ($element, $scope, $window) {
+    var content, wrapper;
+    var directiveClassPrefix = 'chr_h-scroll-shadows';
+
+    (function init () {
+      wrapContents();
+      initToggleShadowsHandlers();
+      toggleShadows();
+    }());
+
+    /**
+     * Makes shadow toggle on the wrapper element scrolling,
+     * or DOM changes inside content element, or window resize
```

Comment:
i think the whole difference between using the observer and using `$watch` is worth being documented

Reply:
I split the func into 5 funcs. So now it should be descriptive.